From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Mon, 7 Sep 2026 13:31:37 +0200
Subject: feat(busybox): use busybox-initramfs

Both packages busybox and busybox-static provide the /usr/bin/busybox
binary. Dracut can use the busybox package. But busybox-static is not
supported, because busybox-static is built with
CONFIG_FEATURE_PREFER_APPLETS.

So only check for busybox from busybox-initramfs to avoid using busybox
from busybox-static.

Forwarded: not-needed
---
 modules.d/10busybox/module-setup.sh    | 4 ++--
 modules.d/88squash-lib/module-setup.sh | 2 +-
 test/TEST-15-BUSYBOX/test.sh           | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/modules.d/10busybox/module-setup.sh b/modules.d/10busybox/module-setup.sh
index 25ff0fd..5fbd325 100755
--- a/modules.d/10busybox/module-setup.sh
+++ b/modules.d/10busybox/module-setup.sh
@@ -2,7 +2,7 @@
 
 # called by dracut
 check() {
-    require_binaries busybox || return 1
+    require_binaries /usr/lib/initramfs-tools/bin/busybox || return 1
 
     return 255
 }
@@ -18,7 +18,7 @@ install() {
     local _path _p _busybox _busybox_path
     local _dstdir="${dstdir:-"$initdir"}"
     local _progs=()
-    _busybox=$(find_binary busybox)
+    _busybox=/usr/lib/initramfs-tools/bin/busybox
     _busybox_path="/usr/bin/busybox"
 
     # install busybox symlinks manually
diff --git a/modules.d/88squash-lib/module-setup.sh b/modules.d/88squash-lib/module-setup.sh
index f1d3f1b..e06e383 100755
--- a/modules.d/88squash-lib/module-setup.sh
+++ b/modules.d/88squash-lib/module-setup.sh
@@ -37,7 +37,7 @@ squash_install() {
     # verify that there is a valid handler before doing anything
     squash_get_handler > /dev/null || return 1
 
-    _busybox=$(find_binary busybox)
+    _busybox=/usr/lib/initramfs-tools/bin/busybox
 
     # Create mount points for squash loader and basic directories
     mkdir -p "$initdir"/squash
diff --git a/test/TEST-15-BUSYBOX/test.sh b/test/TEST-15-BUSYBOX/test.sh
index 8004221..f964e5c 100755
--- a/test/TEST-15-BUSYBOX/test.sh
+++ b/test/TEST-15-BUSYBOX/test.sh
@@ -8,7 +8,7 @@ TEST_DESCRIPTION="busybox applets win over host binaries when busybox module is
 #DEBUGFAIL="rd.debug rd.shell"
 
 test_check() {
-    if ! type -p busybox &> /dev/null; then
+    if ! type -p /usr/lib/initramfs-tools/bin/busybox &> /dev/null; then
         echo "Test needs busybox... Skipping"
         return 1
     fi
